src/components/ec-inline-input-field/ec-inline-input-field.vue

```diff
@@ -0,0 +1,101 @@
+<template>
+  <div
+    class="tw-truncate"
+    data-test="ec-inline-input-field"
+  >
+    <div class="tw-mini-header">{{ label }}</div>
+    <div v-if="isEditable">
+      <ec-inline-input-field-read-only
+        v-if="isReadOnly"
+        :value="value"
+        :gain-focus="gainFocus"
+        @edit="edit"
+      />
+      <ec-inline-input-field-edit
+        v-else-if="isEditing"
+        :original-value="value"
+        :status="status"
+        @cancel="cancel"
+        @submit="submit"
+      />
+      <ec-inline-input-field-loading
+        v-else
+        :value="innerValue"
+      />
+    </div>
+    <slot v-else />
+  </div>
+</template>
+
+<script>
+import { EDITING, LOADING, READ_ONLY } from '@/enums/input-status';
+import EcInlineInputFieldEdit from './components/edit';
+import EcInlineInputFieldLoading from './components/loading';
+import EcInlineInputFieldReadOnly from './components/read-only';
+
+export default {
+  name: 'EcInlineInputField',
+  components: {
+    EcInlineInputFieldEdit,
+    EcInlineInputFieldLoading,
+    EcInlineInputFieldReadOnly,
+  },
+  props: {
+    label: {
+      default: '',
+      type: String,
+    },
+    isEditable: {
+      type: Boolean,
+      default: false,
+    },
+    status: {
+      type: String,
+      default: READ_ONLY,
+      validator(value) {
+        return [READ_ONLY, EDITING, LOADING].includes(value);
+      },
+    },
+    value: {
+      default: '',
+      type: String,
+    },
+  },
+  data() {
+    return {
+      innerValue: null,
+      gainFocus: false,
+    };
+  },
+  computed: {
+    isEditing() {
+      return this.isEditable && this.status === EDITING;
+    },
+    isReadOnly() {
+      return this.isEditable && this.status === READ_ONLY;
+    },
+  },
+  watch: {
```

Comment:
This is the reason why I suggested to use v-model. You wouldn't need this intermediate innerValue and this watcher. You shouldn't be updating the innerValue at all. After submit you should just notify the parent via `this.$emit('submit', data.value);` and let the parent update the value and pass it back to this component. v-model is suitable for it.

Reply:
Absolutely right. It makes perfect sense to use `v-model` for a case like this. I'm still not used to thinking about this directive at a custom component level.
Even so, I think I would still need that `innerValue` (I've renamed it to `valueForLoading` for clarity). I mean, initially, I had to include that way of preserving the inner temporal value because of the "loading" mode. When we are in that mode we need to show what the user wrote when they were in edit mode without changing really the global `value`, since it is the parent's job, so I think I need that data to show it in the "loading" mode during the time the `submit` was emitted and we are waiting for the parent to save the new `value` or discard it.